src/pages/lessons/[slug].tsx

```diff
@@ -136,8 +137,25 @@ const Lesson: FunctionComponent<LessonProps> = ({initialLesson}) => {
   const [playerVisible, setPlayerVisible] = React.useState<boolean>(false)
   const [lessonView, setLessonView] = React.useState<any>()
   const [watchCount, setWatchCount] = React.useState<number>(0)
+
   const currentPlayerState = playerState.value as string
 
+  const [isIncomingAnonViewer, setIsIncomingAnonViewer] =
+    React.useState<boolean>(false)
+
+  React.useEffect(() => {
+    const storage = window?.sessionStorage
+    if (!storage) return
+
+    const prevPath = storage.getItem('prevPath')
+
+    if (!prevPath && !subscriber && !viewer) {
```

Comment:
if there is no previous path logged then we know they just dropped into this lesson and they aren't anon if we have a subscriber or viewer